src/tools/generators.ts

```diff
@@ -601,3 +608,275 @@ export async function generateAppIcons(option: `${Platforms}` | "all") {
+  ].filter(Boolean)
+
+  if (messages.length) {
+    validationMessages.push(`⚠️  ignite/templates/splash-screen/logo.png:`, ...messages)
+  }
+
+  return {
+    isValid: !validationMessages.length,
+    messages: validationMessages,
+  }
+}
+
+/**
+ * Generates splash screen for all platforms
+ */
+export async function generateSplashScreen(options: {
+  androidSize: number
+  iosSize: number
+  backgroundColor: string
+}) {
+  const { androidSize, iosSize, backgroundColor } = options || {}
+  const { path, exists, write, find } = filesystem
+  const cwd = process.cwd()
+
+  const inputFilePath = path(templatesDir(), "splash-screen", "logo.png")
+  const expoOutputDirPath = path(cwd, "assets/images")
+  const isExpoOutputDirExists = exists(expoOutputDirPath) === "dir"
+  const bootsplashCliPath = path(
+    cwd,
+    "node_modules/react-native-bootsplash/dist/commonjs/generate.js",
+  )
+  const isBootsplashCliInstalled = exists(bootsplashCliPath) === "file"
+  const optionGenerationSuccesses = []
+
+  async function generateForVanilla(type: "ios" | "android", size: number) {
+    const typeName = { ios: "iOS", android: "Android" }[type]
+
+    const bootsplashOutputPath = {
+      android: path(cwd, "android/app"),
+      ios: (function () {
+        const searchPath = path(cwd, "ios")
+
+        if (!exists(searchPath)) return searchPath
+
+        return (
+          find(searchPath, {
+            directories: true,
+            files: false,
+            matching: "*.xcodeproj",
+            recursive: false,
+          })?.[0] || searchPath
+        )
+      })(),
+    }[type]
+
+    const isBootsplashOutputPathExists = exists(bootsplashOutputPath) === "dir"
+
+    heading(`Generating ${typeName} splash screen...`)
+
+    if (!isBootsplashOutputPathExists) {
+      warning(
+        `⚠️  No output directory found for "${typeName}" at "${bootsplashOutputPath}". Skipping...`,
+      )
+      return
+    }
+
+    if (!isBootsplashCliInstalled) {
+      warning(
+        `⚠️  If you are attempting to generate a splash screen for bare/vanilla ${typeName} react-native project, please install and configure the "react-native-bootsplash" package. Skipping...`,
+      )
+      return
+    }
+
+    const { generate } = require(bootsplashCliPath) || {}
+    const logFn = console.log
+    const outputFileNames: string[] = []
+
+    console.log = function (log: string) {
+      if (typeof log !== "string") return
+
+      const filePathMatch = log.match(/(?:android|ios)\/.*\.(?:png|storyboard|xml)/)
+
+      if (!filePathMatch) return
+
+      outputFileNames.push(`✅ ${filePathMatch[0]}`)
+    }
+
+    try {
+      await generate({
+        android: type === "android" && { sourceDir: bootsplashOutputPath },
+        ios: type === "ios" && { projectPath: bootsplashOutputPath },
+        workingPath: cwd,
+        logoPath: inputFilePath,
+        assetsPath: null,
+        backgroundColor,
+        flavor: "main",
+        logoWidth: size,
+      })
```

Comment:
this is the `react-native-bootsplash` cli's node script method. this method by default generates assets for all available platforms in one pass. however, since we construct our own output, we call this separately (passing in a falsy value for the platform we don't need).